modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/DefaultComponentParameterAst.java

```diff
@@ -64,7 +65,7 @@ private DefaultComponentParameterAst(String rawValue, Object complexValue, Suppl
       } else if (isEmpty(rawValue)) {
         final Object defaultValue = getModel().getDefaultValue();
         if (isExpression(defaultValue)) {
-          return left((String) defaultValue);
+          return left(extractExpression(defaultValue).get());
```

Comment:
check for optional presense before setting: `extractExpression(rawValue).ifPresent(expression::set)`